pennylane/circuit_graph.py

```diff
@@ -469,9 +470,59 @@ def get_depth(self):
                 self._operation_graph = self._graph.subgraph(
                     list(self._graph.nodes().index(node) for node in self.operations)
                 )
-                self._depth = rx.dag_longest_path_length(self._operation_graph) + 1
+                self._extend_graph(self._operation_graph)
+                self._depth = (
+                    rx.dag_longest_path_length(
+                        self._operation_graph, weight_fn=lambda _, __, w: self._weight_func(w)
```

Comment:
Why `_, __,` needed?

Reply:
The function `dag_longest_path_length` expects its 2nd argument to be a function which takes 3 inputs. Not specifying this will cause an error.